betodealmeida/incubator-superset
# feat: add force option to report screenshots

### SUMMARY


Add a new column `force_screenshot` to the reports/alerts model. For alerts this is always set to true. For reports this is currently set to false, and will be made configurable.

When the flag is true, when generating a screenshot for the alert/report the cache will be bypassed.

Depends on https://github.com/apache/superset/pull/17695.

### MIGRATION

DB migration adds a new column `force_screenshot`, set to true for existing alerts, false otherwise.

Results:

Current: 0.34 s
10+: 0.35 s
100+: 0.39 s
1000+: 0.74 s
10000+: 3.58 s
100000+: 32.59 s
1000000+: 453.84 s

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF


N/A

### TESTING INSTRUCTIONS


1. Create a report and a alert.
2. Run `SELECT force_screenshot FROM report_schedule`, check that it's true for the alert and false for the report.

### ADDITIONAL INFORMATION
